normalize-git-url.js

```diff
@@ -18,7 +18,16 @@ module.exports = function normalize (u) {
 
   var returnedUrl
   if (parsed.pathname.match(/\/?:/)) {
-    returnedUrl = u.replace(/^(?:git\+)?ssh:\/\//, '').replace(/#[^#]*$/, '')
+    if (u.match(/^git\+https?/) && parsed.pathname.match(/\/?:[^0-9]/)) {
+      returnedUrl = u.replace(/^(.*:[^:]+):(.*)/, '$1/$2')
+      console.warn('deprecated: ' + u + ' is using an' +
```

Comment:
I'm of mixed feelings– I'd love to see a better validation mechanic for the cli (`normalize-package-data` is annoying both due to the fact that it at its core is a data fixer AND because its oridinarily invoked via `read-package-json`)– but delaying a _bug fix_ for architecture ... doesn't pass my cost/benefit balance. And the alternative, adding a regex check directly in the caching code, feels like we're moving backwards here. Long term having the caching code emit deprecation warnings feels ... very poorly factored to me. What was the value in pulling this out in the first place if we then have to copy arbitrary bits of it back…